public/app/features/query/state/queryAnalytics.ts

```diff
@@ -28,6 +36,8 @@ export function emitDataRequestEvent(datasource: DataSourceApi) {
       dashboardId: data.request.dashboardId,
       dataSize: 0,
       duration: data.request.endTime! - data.request.startTime,
+      totalQueries: Object.keys(queryCacheStatus).length,
+      cachedQueries: Object.values(queryCacheStatus).filter((val) => val === true).length,
```

Comment:
Nitpick: I generally don't like to do calculations during assignment unless it's necessary for performance. You might consider declaring totalQueries and cachedQueries and adding them to eventData rather than doing the calculation inline.

Reply:
For me it comes down to the complexity of the calculation. Duration is a very simple calculation that IMO is quite straight forward and doesn't require much prior knowledge to understand intent whereas queryCacheStatus is calculated in the block above. By putting the calculations near that block it makes it easy for a reviewer/future developer to reason about the intent of what you're doing and then in this block, you're just handling assignment.

In other cases, calculations might be complex enough that they're themselves worth testing before composing the object. In that case I would shift the block above into a function that can be tested and return an object with those values and do the assignment in this block  as well.

Reply:
Thanks for the good explanation 👍